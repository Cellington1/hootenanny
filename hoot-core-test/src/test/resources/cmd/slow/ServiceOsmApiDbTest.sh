#!/bin/bash
set -e

# clean out the database
scripts/SetupOsmApiDB.sh

# setup DB variables for automation
source conf/DatabaseConfig.sh

# setup dirs
rm -rf test-output/cmd/ServiceOsmApiDbTest
mkdir -p test-output/cmd/ServiceOsmApiDbTest

# setup args
export HOOT_OPTS="-D services.db.writer.create.user=true -D services.db.writer.email=OsmApiInsert@hoot.local -D reader.add.source.datetime=false -D services.db.writer.output.id.mappings=test-output/cmd/ServiceOsmApiDbTest/idmaps.xml "

export DB_URL="postgresql://$PGUSER:$PGPASSWORD@$PGHOST:$PGPORT/$PGDATABASE"

# do the write operation
echo "Performing write operation"
hoot convert $HOOT_OPTS test-files/ToyTestA.osm "$DB_URL"
#wait

# dump the database
#echo "Doing postgres dump operation"
pg_dump -U $PGUSER -a > test-output/cmd/ServiceOsmApiDbTest/ToyTestA-dump.sql
#wait

# compare input to dumped
#echo "Doing comparison"
scripts/compareOsmXmlToOsmApiDbDump test-files/ToyTestA.osm test-output/cmd/ServiceOsmApiDbTest/idmaps.xml test-output/cmd/ServiceOsmApiDbTest/ToyTestA-dump.sql

# do the read operation
echo "Performing read operation"
hoot convert -D services.db.writer.email=OsmApiInsert@hoot.local -D services.db.writer.create.user=true -D reader.add.source.datetime=false "$DB_URL" test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm

# create new maps file
scripts/generateIdMapXmlFromOsm test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm test-output/cmd/ServiceOsmApiDbTest/idmaps2.xml

# compare input to dumped
#echo "Doing comparison"
scripts/compareOsmXmlToOsmApiDbDumpWriter test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm test-output/cmd/ServiceOsmApiDbTest/idmaps2.xml test-output/cmd/ServiceOsmApiDbTest/ToyTestA-dump.sql
