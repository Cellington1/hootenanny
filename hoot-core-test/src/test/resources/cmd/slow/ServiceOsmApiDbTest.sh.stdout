Creating libpgosm lib
Installing libpgosm lib
Dropping old osm api db
Creating osm api db
Performing write operation
OSM XML file:		test-files/ToyTestA.osm
PostgreSQL dump file:	test-output/cmd/ServiceOsmApiDbTest/ToyTestA-dump.sql

Validating data: nodes
	current_nodes, current_node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS
	nodes, node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS

Validating data: ways
	current_ways, current_way_tags, current_way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS
	ways, way_tags, way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS

Validating relations: ways
	current_relations, current_relation_tags, current_relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
	relations, relation_tags, relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
Performing read operation
OSM XML file:		test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm
PostgreSQL dump file:	test-output/cmd/ServiceOsmApiDbTest/ToyTestA-dump.sql

Validating data: nodes
	current_nodes, current_node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS
	nodes, node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS

Validating data: ways
	current_ways, current_way_tags, current_way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS
	ways, way_tags, way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS

Validating relations: ways
	current_relations, current_relation_tags, current_relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
	relations, relation_tags, relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
